src/os/bluestore/BlueStore.h

```diff
@@ -1517,6 +1519,76 @@ class BlueStore : public ObjectStore,
     }
   };
 
+  class TransBatch : private std::vector<TransContext*> {
+    typedef std::vector<TransContext*> base_t;
+    TransContext::state_t state;
+
+  public:
+    TransBatch(const TransContext::state_t state)
+      : state(state) {
+    }
+    TransBatch(TransContext* const txc)
+      : base_t({ txc }),
+        state(txc->state) {
+    }
+    TransBatch(const TransContext::state_t state,
+               std::deque<TransContext*>&& txc_queue)
+      : state(state) {
+      auto& base_vec = static_cast<base_t&>(*this);
+      base_vec.reserve(txc_queue.size());
+      std::move(std::begin(txc_queue), std::end(txc_queue),
+                std::back_inserter(base_vec));
+    }
+
+    ~TransBatch() = default;
+
+    using base_t::emplace_back;
+    using base_t::push_back;
+    using base_t::front;
+    using base_t::empty;
+    using base_t::size;
+    using base_t::cbegin;
+    using base_t::cend;
+
+    using base_t::begin;
+    using base_t::end;
+
+    TransContext::state_t get_state() const {
+      return state;
+    }
+
+    void set_state(const TransContext::state_t new_state) {
+      for (auto txc : *this) {
+        txc->state = new_state;
+      }
+      state = new_state;
+    }
+
+    TransBatch dissect(std::function<bool(TransContext*)> pred) {
+      TransBatch dissected(get_state());
+
+      for (auto it = this->begin(); it != this->end(); /* see body */) {
```

Comment:
@amitkumar50 Any of those cosmetic changes can be made. However, this PR was made half a year ago. I am sure that there are more several problems with current master.